Fetch feature is broken since 1.0.6

Build fails with:

```
error[E0277]: the `?` operator can only be applied to values that implement `Try`
   --> C:\Users\<user>\.cargo\registry\src\index.crates.io-6f17d22bba15001f\headless_chrome-1.0.6\src\browser\fetcher.rs:389:13
    |
389 |             archive_name(revision.as_ref())?
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `&str`
    |
    = help: the trait `Try` is not implemented for `&str`

error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
   --> C:\Users\<user>\.cargo\registry\src\index.crates.io-6f17d22bba15001f\headless_chrome-1.0.6\src\browser\fetcher.rs:389:44
    |
349 | / fn dl_url<R>(revision: R) -> String
350 | | where
351 | |     R: AsRef<str>,
    | |__________________- this function should return `Result` or `Option` to accept `?`
...
389 |               archive_name(revision.as_ref())?
    |                                              ^ cannot use the `?` operator in a function that returns `std::string::String`
    |
    = help: the trait `FromResidual<_>` is not implemented for `std::string::String`

error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
   --> C:\Users\<user>\.cargo\registry\src\index.crates.io-6f17d22bba15001f\headless_chrome-1.0.6\src\browser\fetcher.rs:412:44
    |
394 | fn archive_name<R: AsRef<str>>(revision: R) -> &'static str {
    | ----------------------------------------------------------- this function should return `Result` or `Option` to accept `?`
...
412 |         if revision.as_ref().parse::<u32>()? > 591_479 {
    |                                            ^ cannot use the `?` operator in a function that returns `&str`
    |
    = help: the trait `FromResidual<Result<Infallible, ParseIntError>>` is not implemented for `&str`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `headless_chrome` (lib) due to 3 previous errors
```

It fails on `1.0.6`, `1.0.7` and I believe also on the current `master` branch.
It works if I build without `fetch` feature or on `1.0.5`.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,7 +25,7 @@ ureq = { version = "2.5", optional = true }
 walkdir = { version = "2", optional = true }
 tungstenite = "0.20"
 url = "2.3"
-which = "4.0"
+which = "5.0"
 zip = { version = "0.6.3", optional = true }
 
 [target.'cfg(windows)'.dependencies]

--- a/src/browser/fetcher.rs
+++ b/src/browser/fetcher.rs
@@ -366,7 +366,7 @@ where
             "{}/chromium-browser-snapshots/Mac/{}/{}.zip",
             DEFAULT_HOST,
             revision.as_ref(),
-            archive_name(revision.as_ref())?
+            archive_name(revision.as_ref())
         )
     }
 
@@ -376,7 +376,7 @@ where
             "{}/chromium-browser-snapshots/Mac_Arm/{}/{}.zip",
             DEFAULT_HOST,
             revision.as_ref(),
-            archive_name(revision.as_ref())?
+            archive_name(revision.as_ref())
         )
     }
 
@@ -386,7 +386,7 @@ where
             "{}/chromium-browser-snapshots/Win_x64/{}/{}.zip",
             DEFAULT_HOST,
             revision.as_ref(),
-            archive_name(revision.as_ref())?
+            archive_name(revision.as_ref())
         )
     }
 }
@@ -409,7 +409,7 @@ fn archive_name<R: AsRef<str>>(revision: R) -> &'static str {
     #[cfg(windows)]
     {
         // Windows archive name changed at r591479.
-        if revision.as_ref().parse::<u32>()? > 591_479 {
+        if revision.as_ref().parse::<u32>() > Ok(591_479) {
             "chrome-win"
         } else {
             "chrome-win32"

